src/System.Text.Json/src/System/Text/Json/Reader/JsonReaderHelper.cs

```diff
@@ -263,8 +263,12 @@ public static bool TryGetEscapedGuid(ReadOnlySpan<byte> span, out Guid value)
             utf8Unescaped = utf8Unescaped.Slice(0, written);
             Debug.Assert(!utf8Unescaped.IsEmpty);
 
-            value = default;
-            return (utf8Unescaped.Length == JsonConstants.MaximumFormatGuidLength) && Utf8Parser.TryParse(utf8Unescaped, out value, out _, 'D');
+            if (utf8Unescaped.Length != JsonConstants.MaximumFormatGuidLength)
```

Comment:
Now we have an additional if/branch.

I would have kept the previous change (since the setting the `value` to default is cheaper than adding branches)

Reply:
True. In https://github.com/dotnet/corefx/pull/36347, I've changed it back to the original.